webui/src/lib/components/repository/changes.jsx

```diff
@@ -280,3 +283,48 @@ const ExperimentalDeltaDiffButton = ({showButton = false}) => {
         </ExperimentalOverlayTooltip>
     </>
 }
+
+export const MetadataFields = ({ metadataFields, setMetadataFields}) => {
+    return (
+        <div className="mt-3">
+            {metadataFields.map((f, i) => {
+                return (
+                    <Form.Group key={`commit-metadata-field-${f.key}-${f.value}-${i}`} className="mb-3">
+                        <Row>
+                            <Col md={{span: 5}}>
+                                <Form.Control type="text" placeholder="Key" defaultValue={f.key}
+                                              onChange={(e) => {
```

Comment:
You shouldn't mutate props like this. A good way of thinking about props is as read-only. Want to make a change? Create a new one with your changes and pass that in as the parameters to the setter function. If you need to do multiple levels of spread and it gets messy, you can use something like [Immer](https://immerjs.github.io/immer/), but I'm not sure that's needed at this point.  
While you're at it, maybe move that anonymous function to a named function with `useCallback`? 😉

Reply:
Honestly I just moved existing code and wrapped it in its own component - didn't really dig into how it works. I'll give it a go.

Reply:
> Honestly I just moved existing code and wrapped it in its own component. 

Sorry, did not consider that. Thanks for the fix!